### PR TITLE
[visualization] remove warnings + minor CMakeLists fix

### PR DESCRIPTION
**Pull request by nzlz**
### Description
Merge master branch and remove various moveit_ros_visualization related warnings.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Document API changes relevant to the user in the moveit/MIGRATION.md notes
- [ ] Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Decide if this should be cherry-picked to other current ROS branches
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by LanderU — CHANGES_REQUESTED** [review]
This PR is huge, can we include only the changes that are needed.

**Comment by nzlz** [comment]
All the commits before a7bbe3b are the _Merge master branch + resolve conflicts_. The goal is to keep the branch up to date to prevent outdated compilation issues etc. I can remove that commit and all the previous and leave only the warning fixes.

**Comment by ahcorde** [comment]
can we create a new PR with only the new changes?

**Comment by nzlz** [comment]
PR updated only with the changes.

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h [hunk 1]
```diff
@@ -50,7 +50,7 @@
 namespace planning_scene_monitor
 {
 
-typedef std::function<void(const sensor_msgs::msg::JointState::ConstPtr& joint_state)> JointStateUpdateCallback;
+typedef std::function<void(const sensor_msgs::msg::JointState::ConstSharedPtr& joint_state)> JointStateUpdateCallback;
```
Comment (ahcorde) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h:53]: Why are you changing pointer to shared?

File: moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp [hunk 2]
```diff
@@ -169,11 +169,6 @@ void RobotStateDisplay::changedEnableCollisionVisible()
   robot_->setCollisionVisible(enable_collision_visible_->getBool());
 }
 
-static bool operator!=(const std_msgs::msg::ColorRGBA& a, const std_msgs::msg::ColorRGBA& b)
```
Comment (ahcorde) [inline moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp:172]: Why are you removing this operator?

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp (hunk 2 above)
Comment (anasarrak) [inline moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp:172]: Why are we removing this?

Comment (anasarrak) [reply moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp:172]: https://github.com/AcutronicRobotics/moveit2/pull/127/files/c0f4e01e76871b9f158be5600ff8562005465fe4#r302404706

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h (hunk 1 above)
Comment (nzlz) [reply moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h:53]: ConstPtr is deprecated and throws warnings. As why it should be translated to ConstSharedPtr, if we have no reason to make it unique or weak, should be shared. Check slides 7-8 of https://roscon.ros.org/2015/presentations/state-of-ros2.pdf .

File: moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp (hunk 2 above)
Comment (nzlz) [reply moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp:172]: Its duplicated with the ColorRGBA message operator, which already contains the =! operator. So its not needed + throws a warning.

**Review by ahcorde — APPROVED** [review]

**Pull request merged**